tests/framework/extensions/provisioning/creates.go

```diff
@@ -239,8 +239,115 @@ func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvide
 		if err != nil {
 			return nil, err
 		}
+	}
+
+	createdCluster, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ByID(namespace + "/" + clusterName)
+	return createdCluster, err
+}
+
+// CreateProvisioningK3SCustomCluster provisions a k3s cluster using a 3rd party client for its nodes, then runs verify checks
+func CreateProvisioningK3SCustomCluster(client *rancher.Client, externalNodeProvider ExternalNodeProvider, clustersConfig *clusters.ClusterConfig) (*v1.SteveAPIObject, error) {
+	setLogrusFormatter()
+	rolesPerNode := []string{}
+	quantityPerPool := []int32{}
+	rolesPerPool := []string{}
+	for _, nodes := range *clustersConfig.NodesAndRoles {
+		var finalRoleCommand string
+		if nodes.ControlPlane {
+			finalRoleCommand += " --controlplane"
+		}
+		if nodes.Etcd {
+			finalRoleCommand += " --etcd"
+		}
+		if nodes.Worker {
+			finalRoleCommand += " --worker"
+		}
+		if nodes.Windows {
```

Comment:
if we were to separate these, you would need to remove windows from k3s.

Reply:
Resolved, this comment is no longer relevant as I removed this function. RKE2/K3S are sharing a function again to make it simpler.